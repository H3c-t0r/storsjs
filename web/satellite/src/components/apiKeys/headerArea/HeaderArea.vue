// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="api-keys-header-container">
        <div class="api-keys-header-container__item">
            <SearchArea />
            <Button label="New API Key" width="240px" height="58px" :onPress="onDeleteButtonClick" />
        </div>
        <div class="api-keys-header-container__item">
            <SortApiKeysHeader />
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import SearchArea from './SearchArea.vue';
import Button from '@/components/common/Button.vue';
import SortApiKeysHeader from '@/components/apiKeys/headerArea/SortApiKeysHeader.vue';

@Component({
    components: {
        SearchArea,
        Button,
        SortApiKeysHeader,
    }
})

export default class HeaderArea extends Vue {
}
</script>

<style scoped lang="scss">
    .api-keys-header-container {

        &__item {
            display: flex;
            align-items: center;
            justify-content: space-between;

            &:last-child {
                padding-left: 20px;
            }
        }
    }
</style>