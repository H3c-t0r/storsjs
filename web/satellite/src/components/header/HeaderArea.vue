// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template src="./headerArea.html"></template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import NewProjectArea from '@/components/header/NewProjectArea.vue';
import ProjectSelectionArea from '@/components/header/projectSelection/ProjectSelectionArea.vue';
import NavigationArea from '@/components/navigation/NavigationArea.vue';
import ProjectCreationSuccessPopup from '@/components/project/ProjectCreationSuccessPopup.vue';

import AccountButton from './AccountButton.vue';

@Component({
    components: {
        ProjectCreationSuccessPopup,
        ProjectSelectionArea,
        NewProjectArea,
        AccountButton,
        NavigationArea,
    },
})
export default class DashboardHeader extends Vue {
    public isNavigationVisible: boolean = false;

    public toggleNavigationVisibility(): void {
        this.isNavigationVisible = !this.isNavigationVisible;
    }

    public onLogoClick(): void {
        location.reload();
    }
}
</script>

<style src="./headerArea.scss" scoped lang="scss"></style>
