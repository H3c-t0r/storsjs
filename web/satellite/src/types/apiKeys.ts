// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all apiKey-related functionality
 */
export interface ApiKeysApi {
    /**
     * Fetch apiKey
     *
     * @returns ApiKey
     * @throws Error
     */
    get(projectId: string): Promise<ApiKey[]>;
    create(projectId: string, name: string): Promise<ApiKey>;
    delete(ids: string[]): Promise<null>;
}

/**
 * ApiKey class holds info for ApiKey entity.
 */
export class ApiKey {
    public id: string;
    public secret: string;
    public name: string;
    public createdAt: string;
    public isSelected: boolean = false;

    constructor(id: string, name: string, createdAt: string, secret: string) {
        this.id = id || '';
        this.name = name || '';
        this.createdAt = createdAt || '';
        this.secret = secret || '';

        this.isSelected = false;
    }

    public formattedName(): string {
        let name = this.name;

        if (name.length < 12) {
            return name;
        }

        return name.slice(0, 12) + '...';
    }

    public getDate(): string {
        if (!this.createdAt) {
            return '';
        }

        return new Date(this.createdAt).toLocaleDateString();
    }
}
