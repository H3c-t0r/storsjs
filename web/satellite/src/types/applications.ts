// Copyright (C) 2024 Storj Labs, Inc.
// See LICENSE for copying information.

import VeeamIcon from '@/assets/apps/veeam.png';
import TrueNASIcon from '@/assets/apps/truenas.png';
import SplunkIcon from '@/assets/apps/splunk.png';
import ArqIcon from '@/assets/apps/arq.png';
import ZertoIcon from '@/assets/apps/zerto.png';
import UnitrendsIcon from '@/assets/apps/unitrends.png';
import S3FSIcon from '@/assets/apps/s3fs.png';
import RucioIcon from '@/assets/apps/rucio.png';
import PhotosPlusIcon from '@/assets/apps/photos+.png';
import MSP360Icon from '@/assets/apps/msp360.png';
import LucidLinkIcon from '@/assets/apps/lucidlink.png';
import IconikIcon from '@/assets/apps/iconik.png';
import HammerspaceIcon from '@/assets/apps/hammerspace.png';
import ElementsIcon from '@/assets/apps/elements.png';
import DockerIcon from '@/assets/apps/docker.png';
import PixelfedIcon from '@/assets/apps/pixelfed.png';
import MastodonIcon from '@/assets/apps/mastodon.png';
import DataverseIcon from '@/assets/apps/dataverse.png';
import CyberDuckIcon from '@/assets/apps/cyberduck.png';
import AtempoIcon from '@/assets/apps/atempo.png';
import AcronisIcon from '@/assets/apps/acronis.png';
import MountainDuckIcon from '@/assets/apps/mountainduck.png';
import FileZillaIcon from '@/assets/apps/filezilla.svg';
import GBLabsIcon from '@/assets/apps/gblabs.png';
import AdSignalIcon from '@/assets/apps/adsignal.png';
import OwnCloudIcon from '@/assets/apps/owncloud.png';
import LivepeerIcon from '@/assets/apps/livepeer.png';
import BunnyCDNIcon from '@/assets/apps/bunnycdn.png';
import CometIcon from '@/assets/apps/comet.png';
import FastlyIcon from '@/assets/apps/fastly.jpg';
import GlobusIcon from '@/assets/apps/globus.png';
import HuggingFaceIcon from '@/assets/apps/huggingface.svg';
import KerberosIcon from '@/assets/apps/kerberos.png';

export enum AppCategory {
    All = 'All',
    FileManagement = 'File Management',
    BackupRecovery = 'Backup & Recovery',
    ContentDistribution = 'Content Distribution',
    CDN = 'CDN',
    AI = 'AI',
    Logs = 'Logs',
    SocialMedia = 'Social Media',
}

export type Application = {
    title: string
    description: string
    category: AppCategory
    src: string
    docs: string
}

export const applications: Application[] = [
    {
        title: 'TrueNAS - iX Systems',
        description: 'Back up a single instance or keep several TrueNAS deployments synchronized using a single source of truth.',
        category: AppCategory.BackupRecovery,
        src: TrueNASIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/ix-systems-truenas',
    },
    {
        title: 'Nebula by GB Labs',
        description: 'Cloud-based media storage solution powered by Storj. Work directly from the GB Labs Cloud with project files that are always in sync.',
        category: AppCategory.FileManagement,
        src: GBLabsIcon,
        docs: 'https://www.storj.io/partner-solutions/gb-labs',
    },
    {
        title: 'Hammerspace',
        description: 'Hammerspace and Storj bring more flexibility and agility to your global data landscape.',
        category: AppCategory.FileManagement,
        src: HammerspaceIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/hammerspace',
    },
    {
        title: 'Ad Signal Match',
        description: 'Deduplicate assets and save on AI costs with Ad Signal Match. Transfer only your unique content to Storj.',
        category: AppCategory.FileManagement,
        src: AdSignalIcon,
        docs: 'https://www.storj.io/partner-solutions/ad-signal',
    },
    {
        title: 'OwnCloud',
        description: 'oCIS, or ownCloud Infinite Scale, is a cutting-edge technology platform for building cloud-native file sync and share applications.',
        category: AppCategory.FileManagement,
        src: OwnCloudIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/ocis',
    },
    {
        title: 'Livepeer',
        description: 'Add live and on-demand video experiences by transcoding videos with Livepeer and storing your media with Storj.',
        category: AppCategory.BackupRecovery,
        src: LivepeerIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/livepeer',
    },
    {
        title: 'Acronis',
        description: 'Reliable backup and disaster recovery solutions for data archiving and organization, seamlessly integrating with Storj.',
        category: AppCategory.BackupRecovery,
        src: AcronisIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/acronis',
    },
    {
        title: 'Iconik',
        description: 'Iconik is easy to use and intuitive with a clean and intuitive platform that organizes your media and makes it searchable.',
        category: AppCategory.FileManagement,
        src: IconikIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/iconik',
    },
    {
        title: 'Photos+',
        description: 'Store and manage your photos and videos in your own cloud storage account.',
        category: AppCategory.FileManagement,
        src: PhotosPlusIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/photos-plus',
    },
    {
        title: 'Veeam',
        description: 'All-in-one backup, recovery, and data security solution that serves both on-premises and cloud storage.',
        category: AppCategory.BackupRecovery,
        src: VeeamIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/veeam',
    },
    {
        title: 'Splunk',
        description: 'Splunk is a data analytics platform that provides data-driven insights across all aspects of a company.',
        category: AppCategory.Logs,
        src: SplunkIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/splunk',
    },
    {
        title: 'Arq Backup',
        description: 'Arq backs up your filesystem with perfect, point-in-time backups of your files.',
        category: AppCategory.BackupRecovery,
        src: ArqIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/arq',
    },
    {
        title: 'Zerto',
        description: 'Easily incorporate immutability into a complete disaster recovery strategy.',
        category: AppCategory.BackupRecovery,
        src: ZertoIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/zerto',
    },
    {
        title: 'Unitrends',
        description: 'Automate manual tasks, eliminate management complexity, and deliver tested hardware and software resilience.',
        category: AppCategory.BackupRecovery,
        src: UnitrendsIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/unitrends',
    },
    {
        title: 'S3FS',
        description: 's3fs allows Linux, macOS, and FreeBSD to mount an S3 bucket via FUSE.',
        category: AppCategory.FileManagement,
        src: S3FSIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/s3fs',
    },
    {
        title: 'Rucio',
        description: 'Organize, manage, and access your data at scale with Rucio, an open-source framework for scientific collaboration.',
        category: AppCategory.FileManagement,
        src: RucioIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/rucio',
    },
    {
        title: 'MSP360',
        description: 'MSP360 is a best-in-class IT management platform for MSPs and internal IT teams.',
        category: AppCategory.BackupRecovery,
        src: MSP360Icon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/msp360',
    },
    {
        title: 'LucidLink',
        description: 'Using Storj with LucidLink provides resilient cloud object storage with blazing performance and zero-trust security.',
        category: AppCategory.FileManagement,
        src: LucidLinkIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/lucidlink',
    },
    {
        title: 'Elements',
        description: 'Store, manage and collaborate on media content using a secure and scalable asset management platform.',
        category: AppCategory.FileManagement,
        src: ElementsIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/elements',
    },
    {
        title: 'Docker',
        description: 'Storj supports custom Content-Type for any key, so it can be used as a container registry to distribute container images.',
        category: AppCategory.ContentDistribution,
        src: DockerIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/docker',
    },
    {
        title: 'Pixelfed',
        description: 'Learn how to set up Pixelfed decentralized social media platform to Storj.',
        category: AppCategory.SocialMedia,
        src: PixelfedIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/pixelfed',
    },
    {
        title: 'Mastodon',
        description: 'Here is how you can set up Mastodon decentralized social media platform to Storj.',
        category: AppCategory.SocialMedia,
        src: MastodonIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/mastodon',
    },
    {
        title: 'Dataverse',
        description: 'S3 credentials allow Dataverse to upload and download files from Storj as if it was using the S3 API.',
        category: AppCategory.FileManagement,
        src: DataverseIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/dataverse',
    },
    {
        title: 'CyberDuck',
        description: 'Open-source cloud storage browser app for macOS, Windows, and Linux that supports Storj.',
        category: AppCategory.FileManagement,
        src: CyberDuckIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/cyberduck',
    },
    {
        title: 'Atempo',
        description: 'Integrate with Storj to reduce storage costs without sacrificing security or performance for media archival with global access.',
        category: AppCategory.BackupRecovery,
        src: AtempoIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/atempo-miria',
    },
    {
        title: 'Mountain Duck',
        description: 'Add Storj as a disk on your computer. Open your Storj files with any application and work like on a local disk.',
        category: AppCategory.FileManagement,
        src: MountainDuckIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/cyberduck',
    },
    {
        title: 'Bunny CDN',
        description: 'Set up a static website hosting with Storj using Bunny CDN as a content delivery network providing a caching layer.',
        category: AppCategory.CDN,
        src: BunnyCDNIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/bunny',
    },
    {
        title: 'Fastly',
        description: 'Distribute your content among the Fastly edge cloud service using your Storj buckets as a source of content.',
        category: AppCategory.CDN,
        src: FastlyIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/fastly',
    },
    {
        title: 'Comet',
        description: 'Flexible backup platform integrates with Storj as a backup and storage destination to protect and restore data.',
        category: AppCategory.BackupRecovery,
        src: CometIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/comet',
    },
    {
        title: 'Globus',
        description: 'Seamless collaboration and management for data transfer, sharing, and discovery, all into one unified open-source platform.',
        category: AppCategory.FileManagement,
        src: GlobusIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/globus',
    },
    {
        title: 'HuggingFace',
        description: 'Train and deploy open-source AI models, while saving and loading datasets to and from Storj.',
        category: AppCategory.AI,
        src: HuggingFaceIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/hugging-face',
    },
    {
        title: 'FileZilla',
        description: 'Learn how to set up FileZilla to transfer files over Storj DCS or integrate FileZilla Pro.',
        category: AppCategory.FileManagement,
        src: FileZillaIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/filezilla',
    },
    {
        title: 'Kerberos.io',
        description: 'Open-source platform for video analytics and management. Integrate with Storj to store your Kerberos Vault video files.',
        category: AppCategory.FileManagement,
        src: KerberosIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/kerberos-vault',
    },
];
