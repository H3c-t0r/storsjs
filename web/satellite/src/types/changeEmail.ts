// Copyright (C) 2024 Storj Labs, Inc.
// See LICENSE for copying information.

export enum ChangeEmailStep {
    InitStep = 0,
    VerifyPasswordStep,
    Verify2faStep,
    VerifyOldEmailStep,
    SetNewEmailStep,
    VerifyNewEmailStep,
    SuccessStep,
}
