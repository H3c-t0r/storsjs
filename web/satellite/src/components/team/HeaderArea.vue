// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="team-header-container">
	    <h1>Project Members</h1>
	    <div class="team-header-container__wrapper">
            <HeaderComponent ref="headerComponent" placeHolder="Team Members" :search="processSearchQuery">
                <div class="header-default-state" v-if="headerState === 0">
                    <span>The only project role currently available is Admin, which gives <b>full access</b> to the project.</span>
                    <Button class="button" label="+Add" width="122px" height="48px" :onPress="onAddUsersClick"/>
                </div>
                <div class="header-selected-members" v-if="headerState === 1 && !isDeleteClicked">
                    <Button class="button deletion" label="Delete" width="122px" height="48px" :onPress="onFirstDeleteClick"/>
                    <Button class="button" label="Cancel" width="122px" height="48px" isWhite="true" :onPress="onClearSelection"/>
                </div>
                <div class="header-after-delete-click" v-if="headerState === 1 && isDeleteClicked">
                    <span>Are you sure you want to delete {{selectedProjectMembersCount}} {{userCountTitle}}?</span>
                    <div class="header-after-delete-click__button-area">
                        <Button class="button deletion" label="Delete" width="122px" height="48px" :onPress="onDelete"/>
                        <Button class="button" label="Cancel" width="122px" height="48px" isWhite="true" :onPress="onClearSelection"/>
                    </div>
                </div>
            </HeaderComponent>
            <div class="blur-content" v-if="isDeleteClicked"></div>
            <div class="blur-search" v-if="isDeleteClicked"></div>
	    </div>
        <AddUserPopup v-if="isAddTeamMembersPopupShown"/>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import Button from '@/components/common/Button.vue';
import HeaderComponent from '@/components/common/HeaderComponent.vue';
import AddUserPopup from '@/components/team/AddUserPopup.vue';

import { ProjectMember, ProjectMemberHeaderState } from '@/types/projectMembers';
import { APP_STATE_ACTIONS, NOTIFICATION_ACTIONS, PM_ACTIONS } from '@/utils/constants/actionNames';

declare interface ClearSearch {
    clearSearch(): void;
}

@Component({
    components: {
        Button,
        HeaderComponent,
        AddUserPopup,
    },
})
export default class HeaderArea extends Vue {
    @Prop({default: ProjectMemberHeaderState.DEFAULT})
    private readonly headerState: ProjectMemberHeaderState;
    @Prop({default: 0})
    public readonly selectedProjectMembersCount: number;

    private FIRST_PAGE = 1;

    public isDeleteClicked: boolean = false;

    public $refs!: {
        headerComponent: HeaderComponent & ClearSearch;
    };

    public get userCountTitle(): string {
        if (this.selectedProjectMembersCount === 1) {
            return 'user';
        }

        return 'users';
    }

    public onAddUsersClick(): void {
        this.$store.dispatch(APP_STATE_ACTIONS.TOGGLE_TEAM_MEMBERS);
    }

    public onFirstDeleteClick(): void {
        this.isDeleteClicked = true;
    }

    public onClearSelection(): void {
        this.$store.dispatch(PM_ACTIONS.CLEAR_SELECTION);
        this.isDeleteClicked = false;

        this.$refs.headerComponent.clearSearch();
    }

    public async onDelete(): Promise<void> {
        const projectMemberEmails: string[] = this.$store.getters.selectedProjectMembers.map((member: ProjectMember) => {
            return member.user.email;
        });

        try {
            await this.$store.dispatch(PM_ACTIONS.DELETE, projectMemberEmails);
        } catch (err) {
            this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, `Error while deleting users from projectMembers. ${err.message}`);

            return;
        }

        this.$store.dispatch(NOTIFICATION_ACTIONS.SUCCESS, 'Members was successfully removed from project');
        this.isDeleteClicked = false;

        this.$refs.headerComponent.clearSearch();
    }

    public async processSearchQuery(search: string): Promise<void> {
        this.$store.dispatch(PM_ACTIONS.SET_SEARCH_QUERY, search);
        try {
            await this.$store.dispatch(PM_ACTIONS.FETCH, this.FIRST_PAGE);
        } catch (err) {
            this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, `Unable to fetch project members. ${err.message}`);
        }
    }

    public get isAddTeamMembersPopupShown(): boolean {
        return this.$store.state.appStateModule.appState.isAddTeamMembersPopupShown;
    }
}
</script>

<style scoped lang="scss">
	h1 {
		font-family: 'font_bold';
		font-size: 32px;
		line-height: 39px;
		margin: 0;
	}

    .header-default-state,
    .header-after-delete-click {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        height: 85px;

        span {
            font-family: 'font_medium';
            font-size: 14px;
            line-height: 28px;
        }

        &__button-area {
            display: flex;

            .deletion {
                margin-top: 2px;
            }
        }
    }

    .header-selected-members {
        display: flex;
        align-items: flex-end;
        height: 85px;
        justify-content: center;
    }

    .button {
        margin-right: 12px;
    }

    span {
        font-family: 'font_regular';
        font-size: 14px;
        line-height: 28px;
    }

    .team-header-container__wrapper {
        margin-bottom: 4px;
        display: flex;
        align-items: center;
        justify-content: flex-start;
        position: relative;

        .blur-content {
            position: absolute;
            top: 100%;
            left: 0;
            background-color: #F5F6FA;
            width: 100%;
            height: 70vh;
            z-index: 100;
            opacity: 0.3;
        }

        .blur-search {
            position: absolute;
            bottom: 0;
            right: 0;
            width: 602px;
            height: 56px;
            z-index: 100;
            opacity: 0.3;
            background-color: #F5F6FA;
        }
    }

    .container.deletion {
        background-color: #FF4F4D;

        &.label {
            color: #FFFFFF;
        }

        &:hover {
            background-color: #DE3E3D;
            box-shadow: none;
        }
    }
</style>
