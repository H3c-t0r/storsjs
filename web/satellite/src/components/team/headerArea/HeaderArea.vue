// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="team-header-container">
        <HeaderComponent ref="headerComponent" placeHolder="Team Members" :search="processSearchQuery" title="Project Members">
            <div class="header-default-state" v-if="headerState === 0">
                <Button class="button" label="+Add" width="122px" height="48px" :onPress="onAddUsersClick"/>
                <span>The only project role currently available is Admin, which gives <b>full access</b> to the project.</span>
            </div>
            <div class="header-selected-members" v-if="headerState === 1 && !isDeleteClicked">
                <Button class="button deletion" label="Delete" width="122px" height="48px" :onPress="onFirstDeleteClick"/>
                <Button class="button" label="Cancel" width="122px" height="48px" isWhite="true" :onPress="onClearSelection"/>
            </div>
            <div class="header-after-delete-click" v-if="headerState === 1 && isDeleteClicked">
                <Button class="button deletion" label="Delete" width="122px" height="48px" :onPress="onDelete"/>
                <Button class="button" label="Cancel" width="122px" height="48px" isWhite="true" :onPress="onClearSelection"/>
                <span>Are you sure you want to delete {{selectedProjectMembers}} {{customUserCount}}</span>
            </div>
        </HeaderComponent>
    </div>
</template>

<script lang="ts">
    import { Component, Prop, Vue } from 'vue-property-decorator';
    import Button from '@/components/common/Button.vue';
    import { APP_STATE_ACTIONS, NOTIFICATION_ACTIONS, PM_ACTIONS } from '@/utils/constants/actionNames';
    import HeaderComponent from '@/components/common/HeaderComponent.vue';
    import { TeamMember } from '@/types/teamMembers';
    import { RequestResponse } from '@/types/response';

    declare interface ClearSearch {
        clearSearch: () => void;
    }

    @Component({
        components: {
            Button,
            HeaderComponent,
        }
    })
    export default class HeaderArea extends Vue {
        @Prop({default: 0})
        private readonly headerState: number;
        @Prop({default: 0})
        private readonly selectedProjectMembers: number;

        private isDeleteClicked: boolean = false;

        public $refs!: {
            headerComponent: HeaderComponent & ClearSearch
        };

        public get customUserCount(): string {
            if (this.selectedProjectMembers === 1) {
                return 'user';
            }

            return 'users';
        }

        public onAddUsersClick(): void {
            this.$store.dispatch(APP_STATE_ACTIONS.TOGGLE_TEAM_MEMBERS);
        }

        public onFirstDeleteClick(): void {
            this.isDeleteClicked = true;
        }

        public onClearSelection(): void {
            this.$store.dispatch(PM_ACTIONS.CLEAR_SELECTION);
            this.isDeleteClicked = false;

            this.$refs.headerComponent.clearSearch();
        }

        public async onDelete(): Promise<void> {
            const projectMemberEmails = this.$store.getters.selectedProjectMembers.map((member: TeamMember) => {
                return member.user.email;
            });

            const response = await this.$store.dispatch(PM_ACTIONS.DELETE, projectMemberEmails);

            if (!response.isSuccess) {
                this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, 'Error while deleting users from team');

                return;
            }

            this.$store.dispatch(NOTIFICATION_ACTIONS.SUCCESS, 'Members was successfully removed from project');
            this.isDeleteClicked = false;

            this.$refs.headerComponent.clearSearch();
        }

        public async processSearchQuery(search: string) {
            this.$store.dispatch(PM_ACTIONS.SET_SEARCH_QUERY, search);
            const response: RequestResponse<object> = await this.$store.dispatch(PM_ACTIONS.FETCH);

            if (!response.isSuccess) {
                this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, 'Unable to fetch project members');
            }
        }
    }
</script>

<style scoped lang="scss">
    .header-default-state,
    .header-selected-members,
    .header-after-delete-click {
        display: flex;
        align-items: center;
    }

    .button {
        margin-right: 12px;
    }

    span {
        font-family: 'font_regular';
        font-size: 14px;
        line-height: 28px;
    }

    .team-header-container {
        margin-bottom: 4px;
        display: flex;
        align-items: center;
        justify-content: flex-start;
    }

    .container.deletion {
        background-color: #FF4F4D;

        &.label {
            color: #FFFFFF;
        }

        &:hover {
            background-color: #DE3E3D;
            box-shadow: none;
        }
    }
</style>
