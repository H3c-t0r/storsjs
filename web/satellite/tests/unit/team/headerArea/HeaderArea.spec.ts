// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { createLocalVue, mount, shallowMount } from '@vue/test-utils';
import * as sinon from 'sinon';
import Vuex from 'vuex';
import HeaderArea from '@/components/team/headerArea/HeaderArea.vue';
import { TeamMember } from '@/types/teamMembers';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('appState Actions', () => {
    let store;
    let actions;

    beforeEach(() => {
        actions = {
            toggleAddTeamMembersPopup: jest.fn()
        };

        store = new Vuex.Store({
            modules: {
                appStateModule: {
                    actions
                }
            }
        });
    });

    it('action on onAddUsersClick works correctly', () => {
        const wrapper = mount(HeaderArea, { store, localVue });

        wrapper.vm.onAddUsersClick();

        expect(actions.toggleAddTeamMembersPopup.mock.calls).toHaveLength(1);
    });
});

describe('projectMembers/notification Actions', () => {
    let store;
    let actions;
    let getters;
    let state;
    let teamMember = new TeamMember('test', 'test', 'test@test.test', 'test');
    let teamMember1 = new TeamMember('test1', 'test1', 'test1@test.test', 'test1');
    let searchQuery = 'test';

    beforeEach(() => {
        getters = {
            selectedProjectMembers: () => [teamMember, teamMember1]
        };

        state = {
            searchParameters: {
                searchQuery: ''
            },
        };

        actions = {
            clearProjectMemberSelection: jest.fn(),
            deleteProjectMembers: async () => {
                return {
                    errorMessage: '',
                    isSuccess: true,
                    data: null
                };
            },
            fetchProjectMembers: async () => {
                return {
                    errorMessage: '',
                    isSuccess: true,
                    data: [teamMember, teamMember1]
                };
            },
            setProjectMembersSearchQuery: () => {
                state.searchParameters.searchQuery = searchQuery;
            },
            success: jest.fn()
        };

        store = new Vuex.Store({
            modules: {
                module: {
                    actions,
                    getters,
                    state
                },
            }
        });
    });

    it('action on onClearSelection works correctly', () => {
        let clearSearchSpy = sinon.spy();

        const wrapper = mount(HeaderArea, { store, localVue });

        wrapper.vm.$refs.headerComponent.clearSearch = clearSearchSpy;
        wrapper.vm.onClearSelection();

        expect(actions.clearProjectMemberSelection.mock.calls).toHaveLength(1);
        expect(wrapper.vm.$data.isDeleteClicked).toBe(false);
        expect(clearSearchSpy.callCount).toBe(1);
    });

    it('actions on onDelete works correctly', async () => {
        let clearSearchSpy = sinon.spy();

        const wrapper = mount(HeaderArea, {
            store,
            localVue
        });

        wrapper.vm.$data.headerState = 1;
        wrapper.vm.$data.isDeleteClicked = true;
        wrapper.vm.$refs.headerComponent.clearSearch = clearSearchSpy;

        await wrapper.vm.onDelete();

        const projectMembersEmails = await store.getters.selectedProjectMembers.map((member) => {
            return member.user.email;
        });

        expect(projectMembersEmails).toHaveLength(2);
        expect(actions.deleteProjectMembers).resolves;
        expect(actions.success.mock.calls).toHaveLength(1);
        expect(wrapper.vm.$data.isDeleteClicked).toBe(false);
        expect(clearSearchSpy.callCount).toBe(1);
    });

    it('action on processSearchQuery works correctly', async () => {
        const wrapper = mount(HeaderArea, {
            store,
            localVue
        });

        await wrapper.vm.processSearchQuery(searchQuery);

        expect(state.searchParameters.searchQuery).toMatch('test');
        expect(actions.fetchProjectMembers).resolves;
    });
});

describe('error conditions', () => {
    let store;
    let actions;
    let getters;
    let searchQuery = 'test';

    beforeEach(() => {
        getters = {
            selectedProjectMembers: () => []
        };

        actions = {
            deleteProjectMembers: async () => {
                return {
                    errorMessage: '',
                    isSuccess: false,
                    data: null
                };
            },
            fetchProjectMembers: async () => {
                return {
                    errorMessage: '',
                    isSuccess: false,
                    data: []
                };
            },
            error: jest.fn()
        };

        store = new Vuex.Store({
            modules: {
                module: {
                    actions,
                    getters
                }
            }
        });
    });

    it('function customUserCount if there is only 1 selected user', () => {
        const wrapper = mount(HeaderArea, { store, localVue });

        wrapper.vm.$data.selectedProjectMembers = 1;

        expect(wrapper.vm.userCountTitle).toMatch('user');
    });

    it('function onDelete if delete is not successful', async () => {
        const wrapper = mount(HeaderArea, {
            store,
            localVue
        });

        wrapper.vm.$data.headerState = 1;
        wrapper.vm.$data.isDeleteClicked = true;

        await wrapper.vm.onDelete();

        await store.getters.selectedProjectMembers.map((member) => {
            return member.user.email;
        });

        expect(actions.error.mock.calls).toHaveLength(1);
    });

    it('function processSearchQuery if fetch is not successful', async () => {
        const wrapper = mount(HeaderArea, {
            store,
            localVue
        });

        await wrapper.vm.processSearchQuery(searchQuery);

        expect(actions.error.mock.calls).toHaveLength(1);
    });
});

describe('HeaderArea.vue', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount(HeaderArea);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with default props', () => {
        const wrapper = mount(HeaderArea);

        expect(wrapper.vm.$props.headerState).toBe(0);
        expect(wrapper.vm.$props.selectedProjectMembers).toBe(0);
    });

    it('function customUserCount work correctly', () => {
        const wrapper = mount(HeaderArea);

        expect(wrapper.vm.userCountTitle).toMatch('users');
    });

    it('function onFirstDeleteClick work correctly', () => {
        const wrapper = mount(HeaderArea);

        wrapper.vm.onFirstDeleteClick();

        expect(wrapper.vm.$data.isDeleteClicked).toBe(true);
    });

});

