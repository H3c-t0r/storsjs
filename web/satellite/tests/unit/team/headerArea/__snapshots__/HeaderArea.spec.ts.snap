// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderArea.vue renders correctly 1`] = `
<div class="team-header-container">
  <headercomponent-stub placeholder="Team Members" search="function () { [native code] }" title="Project Members">
    <div class="header-default-state">
      <button-stub label="+Add" width="122px" height="48px" onpress="function () { [native code] }" class="button"></button-stub> <span>The only project role currently available is Admin, which gives <b>full access</b> to the project.</span>
    </div>
    <!---->
    <!---->
  </headercomponent-stub>
</div>
`;
